internal/jimm/permissions/access.go

```diff
@@ -0,0 +1,799 @@
+	switch relation {
+	case ofganames.AdministratorRelation:
+		return string(jujuparams.OfferAdminAccess)
+	case ofganames.ConsumerRelation:
+		return string(jujuparams.OfferConsumeAccess)
+	case ofganames.ReaderRelation:
+		return string(jujuparams.OfferReadAccess)
+	default:
+		return ""
+	}
+}
+
+// ToCloudAccessString maps relation to a cloud access string.
+func ToCloudAccessString(relation openfga.Relation) string {
+	switch relation {
+	case ofganames.AdministratorRelation:
+		return "admin"
+	case ofganames.CanAddModelRelation:
+		return "add-model"
+	default:
+		return ""
+	}
+}
+
+// ToModelAccessString maps relation to a model access string.
+func ToModelAccessString(relation openfga.Relation) string {
+	switch relation {
+	case ofganames.AdministratorRelation:
+		return "admin"
+	case ofganames.WriterRelation:
+		return "write"
+	case ofganames.ReaderRelation:
+		return "read"
+	default:
+		return ""
+	}
+}
+
+// ToModelAccessString maps relation to a controller access string.
+func ToControllerAccessString(relation openfga.Relation) string {
+	switch relation {
+	case ofganames.AdministratorRelation:
+		return "superuser"
+	default:
+		return "login"
+	}
+}
+
+// ToCloudRelation returns a valid relation for the cloud. Access level
+// string can be either "admin", in which case the administrator relation
+// is returned, or "add-model", in which case the can_addmodel relation is
+// returned.
+func ToCloudRelation(accessLevel string) (openfga.Relation, error) {
+	switch accessLevel {
+	case "admin":
+		return ofganames.AdministratorRelation, nil
+	case "add-model":
+		return ofganames.CanAddModelRelation, nil
+	default:
+		return ofganames.NoRelation, errors.E("unknown cloud access")
+	}
+}
+
+// ToModelRelation returns a valid relation for the model.
+func ToModelRelation(accessLevel string) (openfga.Relation, error) {
+	switch accessLevel {
+	case "admin":
+		return ofganames.AdministratorRelation, nil
+	case "write":
+		return ofganames.WriterRelation, nil
+	case "read":
+		return ofganames.ReaderRelation, nil
+	default:
+		return ofganames.NoRelation, errors.E("unknown model access")
+	}
+}
+
+// ToOfferRelation returns a valid relation for the application offer.
+func ToOfferRelation(accessLevel string) (openfga.Relation, error) {
+	switch accessLevel {
+	case "":
+		return ofganames.NoRelation, nil
+	case string(jujuparams.OfferAdminAccess):
+		return ofganames.AdministratorRelation, nil
+	case string(jujuparams.OfferConsumeAccess):
+		return ofganames.ConsumerRelation, nil
+	case string(jujuparams.OfferReadAccess):
+		return ofganames.ReaderRelation, nil
+	default:
+		return ofganames.NoRelation, errors.E("unknown application offer access")
+	}
+}
+
+// GetUserControllerAccess returns the user's level of access to the desired controller.
+func (j *permissionManager) GetUserControllerAccess(ctx context.Context, user *openfga.User, controller names.ControllerTag) (string, error) {
```

Comment:
suggestion: keep all the permissionManager method implementations in permissionmanager.go

Reply:
Do you mean just move all the methods in `access.go` to `permissionmanager.go`? What about `relations.go` which contains some more methods but these ones are generic OpenFGA ones.

Reply:
yes, i think it would be much cleaner to have all methods of a type in a single file..